BE/src/main/java/com/example/demo/Service/InvestorService.java

```diff
@@ -0,0 +1,54 @@
+package com.example.demo.Service;
+
+
+import com.example.demo.Model.Company;
+import com.example.demo.Model.Investor;
+import com.example.demo.Model.Transaction;
+import com.example.demo.Persistence.API.RepositoryFactory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+@RequiredArgsConstructor
+public class InvestorService {
+
+    private final RepositoryFactory repositoryFactory;
+
+
+
+    @Transactional
+    public List<Investor> listAllInvestors()
+    {
+        return repositoryFactory.createInvestorRepository().findAll();
+    }
+
+    @Transactional
+    public Optional<Investor> findById(int id)
+    {
```

Comment:
Usually this type of formatting is C / C++ specific. In Java, you usually open the bracket on the same line as the method signature.